### PR TITLE
Support Stage Channels

### PR DESCRIPTION
**Pull request by auguwu**
PR for #153, will be updated once more documentation is out. This is just a blue-print.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -32,8 +32,8 @@ You can join the server below and join in #support under the **Wumpcord** catego
 ```js
 const { Client } = require('wumpcord');
 const client = new Client({
-  token: '',
-  ws: { intents: ['guilds', 'guildMessages'] }
+  intents: ['guilds', 'guildMessages'],
```
Comment (Vendicated) [inline README.md:35]: unrelated to this PR but I think intents should be in SCREAMING_SNAKE_CASE, so in this case GUILDS and GUILD_MESSAGES to indicate that they're constant strings and also for parity with the discord api

Comment (auguwu) [reply README.md:35]: Since `feat/stage` and Wumpcord are gonna be updated as a major release, I can do that. 👍

Comment (hokkqi) [reply README.md:35]: Honestly, yeah. 

I thought of letting them put them in lowercase and just convert them to uppercase but someone can just import the Intents anyways so, yeah. +1

File: src/gateway/WebSocketClient.ts [hunk 2]
```diff
@@ -311,7 +308,7 @@ export default class WebSocketClient<
   }
 
   /**
-   * Returns the shard information
+   * Returns shard info for the bot
    */
   async getShardInfo(): Promise<types.ShardInfo> {
     const data = this.options.shardCount === 'auto' ?
```
Comment (hokkqi) [inline src/gateway/WebSocketClient.ts:314]: Is that `?` supposed to be there and I'm missing some functionality I didn't know TS had or is it a rogue `?`?

Comment (DonovanDMC) [reply src/gateway/WebSocketClient.ts:314]: That's just a ternary spread over multiple lines

Comment (auguwu) [reply src/gateway/WebSocketClient.ts:314]: Basically what Donovan said here, just a ternary to make code slightly more readable

**Pull request merged**